OMEdit/OMEditLIB/OMS/ElementPropertiesDialog.cpp

```diff
@@ -403,3 +405,91 @@ void ElementPropertiesDialog::updateProperties()
   // accept the dialog
   accept();
 }
+
+/*
+ * event filter for mParameterLineEdits and mInputLineEdit
+ * to detect the focus out event, and update
+ * the default start values from modeldesctiption.xml
+ */
+bool ElementPropertiesDialog::eventFilter(QObject *pObject, QEvent *pEvent)
+{
+  QLineEdit *pLineEdit = qobject_cast<QLineEdit*>(pObject);
+
+  if (pLineEdit && pEvent->type() == QEvent::FocusOut) {
+    if (pLineEdit->text().isEmpty()) {
+      // search the lineEdit index in parameters or inputs
+      int parameterIndex = mParameterLineEdits.indexOf(pLineEdit);
+      int inputIndex = mInputLineEdits.indexOf(pLineEdit);
+
+      if (parameterIndex != -1) {
+        QString pParameterLabelText = mParameterLabels.at(parameterIndex)->text();
+        restoreDefaultStartValue(pParameterLabelText, pLineEdit);
+      }
+
+      if (inputIndex != -1) {
+        QString pInputLabelText = mInputLabels.at(inputIndex)->text();
+        restoreDefaultStartValue(pInputLabelText, pLineEdit);
+      }
+    }
+  }
+  return QWidget::eventFilter(pObject, pEvent);
+}
+
+/*
+ * helper function to restore default start values read from modeldescription.xml for fmus
+ * and 0 for other systems
+ */
+void ElementPropertiesDialog::restoreDefaultStartValue(QString pName, QLineEdit * pLineEdit)
+{
+  if (mpComponent->getLibraryTreeItem()->getOMSElement() && mpComponent->getLibraryTreeItem()->getOMSElement()->connectors) {
+    oms_connector_t** pInterfaces = mpComponent->getLibraryTreeItem()->getOMSElement()->connectors;
+    for (int i = 0 ; pInterfaces[i] ; i++) {
+      QString nameStructure = QString("%1.%2").arg(mpComponent->getLibraryTreeItem()->getNameStructure(), QString(pInterfaces[i]->name));
+      if (pInterfaces[i]->causality == oms_causality_parameter && QString(pInterfaces[i]->name)== pName) {
+        OMSProxy::instance()->omsDelete(nameStructure + ":start");
+        restoreDefaultStartValueHelper(pInterfaces[i], nameStructure, pLineEdit);
+        return;
+      } else if (pInterfaces[i]->causality == oms_causality_input && QString(pInterfaces[i]->name)== pName) {
+        OMSProxy::instance()->omsDelete(nameStructure + ":start");
+        restoreDefaultStartValueHelper(pInterfaces[i], nameStructure, pLineEdit);
+        return;
+      }
+    }
```

Comment:
I would probably simplify this as follows:

```C++
    for (int i = 0; pInterfaces[i]; i++) {
      if (QString(pInterfaces[i]->name) == pName) {
        if (oms_causality_parameter == pInterfaces[i]->causality || oms_causality_input == pInterfaces[i]->causality) {
          QString nameStructure = QString("%1.%2").arg(mpComponent->getLibraryTreeItem()->getNameStructure(), QString(pInterfaces[i]->name));
          OMSProxy::instance()->omsDelete(nameStructure + ":start");
          restoreDefaultStartValueHelper(pInterfaces[i], nameStructure, pLineEdit);
          return;
        }
      }
    }
```

Reply:
True and then perhaps this `restoreDefaultStartValueHelper()` is unnecessary. Don't forget the check `QString(pInterfaces[i]->name)== pName`. Actually I prefer it to be `QString(pInterfaces[i]->name).compare(pName) == 0)`

Reply:
yes the helper function is not needed, fixed with https://github.com/OpenModelica/OpenModelica/pull/6857/commits/4404e5dd437c7dfe931bd72afc1b977bb0205238